components/List/tests/snapshot/MDCSelfSizingStereoCellSnapshotTests.m

```diff
@@ -109,78 +124,111 @@ - (void)generateSnapshotWithContentSizeCategoryAndNotificationPost:
-    self.cell.titleLabel.font = self.typographyScheme.subtitle1;
-    self.cell.detailLabel.text = @"Detail";
-    self.cell.detailLabel.font = self.typographyScheme.button;
-    self.cell.mdc_adjustsFontForContentSizeCategory = YES;
-    [self.cell setNeedsLayout];
+    cell.titleLabel.text = @"Title";
+    cell.titleLabel.font = self.typographyScheme.subtitle1;
+    cell.detailLabel.text = @"Detail";
+    cell.detailLabel.font = self.typographyScheme.button;
+    cell.mdc_adjustsFontForContentSizeCategory = YES;
+    self.arrayOfCells = @[ cell ];
 
     // Then
     [self generateSnapshotWithContentSizeCategoryAndNotificationPost:UIContentSizeCategoryExtraSmall
-                                                    andVerifyForView:self.cell];
+                                                    andVerifyForView:self.collectionView];
   }
 }
 
 - (void)testCellWithDynamicTypeForContentSizeCategoryExtraLargeEnabledForTitleAndDetail {
   if (@available(iOS 10.0, *)) {
     // Given
+    MDCSelfSizingStereoCell *cell = [[MDCSelfSizingStereoCell alloc] init];
     self.typographyScheme =
         [[MDCTypographyScheme alloc] initWithDefaults:MDCTypographySchemeDefaultsMaterial201902];
 
     // When
-    self.cell.titleLabel.text = @"Title";
-    self.cell.titleLabel.font = self.typographyScheme.subtitle1;
-    self.cell.detailLabel.text = @"Detail";
-    self.cell.detailLabel.font = self.typographyScheme.button;
-    self.cell.mdc_adjustsFontForContentSizeCategory = YES;
-    [self.cell setNeedsLayout];
+    cell.titleLabel.text = @"Title";
+    cell.titleLabel.font = self.typographyScheme.subtitle1;
+    cell.detailLabel.text = @"Detail";
+    cell.detailLabel.font = self.typographyScheme.button;
+    cell.mdc_adjustsFontForContentSizeCategory = YES;
+    self.arrayOfCells = @[ cell ];
 
     // Then
     [self generateSnapshotWithContentSizeCategoryAndNotificationPost:UIContentSizeCategoryExtraLarge
-                                                    andVerifyForView:self.cell];
+                                                    andVerifyForView:self.collectionView];
   }
 }
 
+- (NSInteger)numberOfSectionsInCollectionView:(UICollectionView *)collectionView {
+  return 1;
+}
+
+- (NSInteger)collectionView:(UICollectionView *)collectionView
+     numberOfItemsInSection:(NSInteger)section {
+  return self.arrayOfCells.count;
+}
+
+- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView
+                  cellForItemAtIndexPath:(NSIndexPath *)indexPath {
+  MDCSelfSizingStereoCell *dequeuedCell =
+      [collectionView dequeueReusableCellWithReuseIdentifier:kSelfSizingStereoCellIdentifier
+                                                forIndexPath:indexPath];
+  // Temporarily add dequeuedCell as a subview of collectionView
+  // to inherit traitCollection. The dequeuedCell will be removed
+  // from parent at the end of this method.
+  [self.collectionView addSubview:dequeuedCell];
```

Comment:
@andrewoverton is going to open an issue to fix it inside MDCSelfSizingStereoCell by adding support for `traitCollectionDidChange:`:)